dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/controller/GaenControllerTest.java

```diff
@@ -996,75 +977,92 @@ public void testEtag() throws Exception {
 		assertEquals(expectedEtag, response.getHeader("etag"));
 
-		insertNKeysPerDayInInterval(14,
-				midnight.minusDays(4),
-				midnight, midnight.minusHours(12));
-				response = mockMvc
-				.perform(get("/v1/gaen/exposed/"
-						+ midnight.minusDays(8).getTimestamp())
-								.header("User-Agent", "MockMVC"))
+		insertNKeysPerDay(midnight, 14, 10, midnight.minusHours(12), false);
+
+		response = mockMvc
+				.perform(
+						get("/v1/gaen/exposed/" + midnight.minusDays(8).getTimestamp()).header("User-Agent", "MockMVC"))
 				.andExpect(status().is2xxSuccessful()).andReturn().getResponse();
 
 		publishedUntil = Long.parseLong(response.getHeader("X-PUBLISHED-UNTIL"));
 		assertTrue(publishedUntil < System.currentTimeMillis(), "Published until must be in the past");
 		assertNotEquals(expectedEtag, response.getHeader("etag"));
 	}
-	
+
 	@Test
 	public void testMalciousTokenFails() throws Exception {
 		var requestList = new GaenRequest();
 		List<GaenKey> exposedKeys = new ArrayList<GaenKey>();
 		requestList.setGaenKeys(exposedKeys);
 		String token = createMaliciousToken(UTCInstant.now().plusMinutes(5));
-		MvcResult response = mockMvc.perform(post("/v1/gaen/exposed")
-				.contentType(MediaType.APPLICATION_JSON).header("Authorization", "Bearer " + token)
-				.header("User-Agent", "MockMVC").content(json(requestList))).andExpect(request().asyncNotStarted()).andExpect(status().is(401)).andReturn();
+		MvcResult response = mockMvc.perform(post("/v1/gaen/exposed").contentType(MediaType.APPLICATION_JSON)
+				.header("Authorization", "Bearer " + token).header("User-Agent", "MockMVC").content(json(requestList)))
+				.andExpect(request().asyncNotStarted()).andExpect(status().is(401)).andReturn();
 		String authenticateError = response.getResponse().getHeader("www-authenticate");
 		assertTrue(authenticateError.contains("Unsigned Claims JWTs are not supported."));
 	}
 
-	private void verifyZipInZipResponse(MockHttpServletResponse response, int expectKeyCount) throws Exception {
+	/**
+	 * Verifies a zip in zip response, that each inner zip is again valid.
+	 * @param response
+	 * @param expectKeyCount
+	 * @param expectedRollingPeriod
+	 * @throws Exception
+	 */
+	private void verifyZipInZipResponse(MockHttpServletResponse response, int expectKeyCount, int expectedRollingPeriod) throws Exception {
 		ByteArrayInputStream baisOuter = new ByteArrayInputStream(response.getContentAsByteArray());
 		ZipInputStream zipOuter = new ZipInputStream(baisOuter);
 		ZipEntry entry = zipOuter.getNextEntry();
-		while(entry != null) {
+		while (entry != null) {
+			ZipInputStream zipInner = new ZipInputStream(new ByteArrayInputStream(zipOuter.readAllBytes()));
+			verifyKeyZip(zipInner, expectKeyCount, expectedRollingPeriod);
 			entry = zipOuter.getNextEntry();
 		}
 	}
 
+	/**
+	 * Verifies a zip response, checks if keys and signature is correct.
+	 * 
+	 * @param response
```

Comment:
The parameters should either be documented or not listed. Like this it just takes place but doesn't help.